content/concepts/kata/translations.md

```diff
@@ -5,73 +5,124 @@ sidebar: explanation-kata
 
 ### Merge Issues
 
-Oftentimes if two translations are created around the same time, they may both update the description. When approving the 2nd translation, the 1st approved translation will have already updated the description. This will cause a merge conflict which will need to be resolved. You can do this by forking the 2nd translation and updating the description. `<<<<<<<<`, `=========` and `>>>>>>>>` merge labels will show up in the description to help identify where the merge issues are.
+Oftentimes if two translations are created around the same time, they may both update the description. When approving the 2nd translation, the 1st approved translation will have already updated the description. This will cause a merge conflict which has to be resolved. You can do this by forking the 2nd translation and updating the description. `<<<<<<<<`, `=========` and `>>>>>>>>` merge labels will show up in the description to help identify where the merge issues are.
 
 Once you have fixed the description merge issues you can publish the translation. If you are not the author then you will need to have someone else verify your changes and approve the translation.
 
-## Rejecting a translation
+### Rejecting a translation
+
+Sometimes the best course of action is to simply reject a pending translation. Some possible reasons could be:
+
+- translation is outdated and kata changed a lot since the translation has been published,
+- translation is of bad quality and fixing it is too much work,
+- some user authored a newer, better translation and they cannot publish it since there can only be one pending translation per language per kata,
+- there's a justified concern that someone might approve a translation of insufficient quality before its issues are fixed.
+
+It's worth nothing that rejected translations are not wasted work. They are still available in the system, can be forked and serve as a base for new translations, or its code can be simply copied and reused.
+
+## Fixing existing translations
 
-## Forking a translation
+Sometimes a kata needs to be fixed due to an issue, be it a bug in the reference solution, typos in the description or the lack of test coverage. Fixes and improvements to kata can be done in one of two ways:
 
-Code Warriors can fork a translation to improve it and help it get approved faster. However they will not receive rank or honor progression, only the original translator will.
+- If a translation was created with translations panel, its most recent version can be forked, modified and published, which then goes through the usual translation review process and waits until another user approves it. However, translations created directly within the kata editor cannot be forked.
+- A translation may be modified directly within the kata editor. This method of fixing kata is discouraged unless absolutely necessary due to its drawbacks:
```

Comment:
about that, I tend more and more to use directly the edit panel.

There is sort of a contradiction between having the right ot reach the edit panel and the idea to only go through the translation panel.

Note there is a major drawback to proceed only through translations, even if everyone sticks to the process: when forking a translation while another one has been approved, merge conflicts will apear too... :/

Reply:
WARNING: it seems I accidentally removed a sentence of the file with this comment... :/ That wasnt supposed to happen

Reply:
Maybe I should not say that this way is discouraged, just that it can cause some other set of problems with outdated forks later.  Will think about it.

Reply:
> when forking a translation while another one has been approved, merge conflicts will apear too... :/

It's true only for description. By forking, you cannot cause merge conflict for code snippets, but it's possible when code snippet is changed with kata editor. Existing forks become outdated, and when forked again, changes introduced with kata editor are either ignored, or cause merge conflict. 

Maybe this needs some explanation with an example:
 - initial version of translation is created (lets call it `Fork1`), it has fixed test `A`.
 - someone notices that test case `B` is missing. `Fork1` is forked into `Fork2`, test `B` is added.
 - someone notices that test case `C` is missing. They edit the kata with kata editor, and add test case `C`.
 - someone notices that test case `D` is missing. They fork `Fork2`, and... what happens with test case `C`? is it missing from the fork, or is it presented as merge conflict? Does anyone know?

Reply:
Honestly, I'd stay very evasive about all of this in the documentation. The main problem is that there is a structural flaw behind all of this. In the end it's calling for a massive change in the workflow. Too much contradiction everywhere

* access to the edit panel while translation needs a delay => slow the bug fixing. Most users won't do bug fixing if their changes are never reviwed/approved
* going through translations, at each approval, the guy who did shit in the first place get points... XD
* impossible to fix the original language through translations

Fundamentally, there is a contradiction between translation process and maintaining kata.

The flaw is really hue, here. @kazk : any idea/feedback about that?
Thinking about something: changing a bit the UI, allowing to fork a language without regarding the origial translation, and importing by default the current version of the kata. Everything would be up to date, then. It needs more thinking about how to handle rejected/archived translations, though. And what to do about the points too. rewarding a bug fix like closing a parenthese in the deescription? 

do we already have an issue about that somewhere? :o

Reply:
While I agree it's a flaw in the process, I still think it's good thing to have it documented, even if it's not intended behavior. That's what documentation is for: it answers questions, and explains how things work. It does not have to explain *why* things work this way, but it makes users aware of what they can expect from the system.
I actually received questions about this aspect (not many, but there were two or three people through the time) and I did not know how to answer. When users become able to create translations and kata, they become curious and want explanations. I think docs should give answers to them, even if the answer is "it works so and so, but it's a bug".

Reply:
:+1:

Reply:
Related issue is https://github.com/codewars/codewars.com/issues/2180 and if you want to discuss it or have some ideas, it would be great.

Reply:
> @hobovsky ([comment](https://github.com/codewars/docs/pull/130#discussion_r514440134)):
> - initial version of translation is created (lets call it Fork1), it has fixed test A.
> -  someone notices that test case B is missing. Fork1 is forked into Fork2, test B is added.
> -  someone notices that test case C is missing. They edit the kata with kata editor, and add test case C.
> -  someone notices that test case D is missing. They fork Fork2, and... what happens with test case C? is it missing from the fork, or is it presented as merge conflict? Does anyone know?

[answer here](https://github.com/codewars/codewars.com/issues/2180#issuecomment-719010602)